backend/src/routers/gameRouter.ts

```diff
@@ -1,33 +1,81 @@
 import { Router } from 'express';
 import { z } from 'zod';
-import { validateRequestBody } from 'zod-express-middleware';
-import { getAccountFromCid } from '../services/accountService.js';
+import {
+	processRequestQuery,
+	validateRequestBody
+} from 'zod-express-middleware';
+import { GammaUser } from '../models/gammaModels.js';
 import {
 	getGameOwnerIdFromCid,
 	getGameOwnerNameFromId,
 	getGameOwnersWithGames
 } from '../services/gameOwnerService.js';
 import {
-	Filter,
 	createGame,
-	filterGames,
-	getAllGames,
-	searchGames,
+	markGameAsPlayed,
 	removeGame,
-	markGameAsPlayed
+	searchAndFilterGames
 } from '../services/gameService.js';
 import { platformExists } from '../services/platformService.js';
-import sendApiValidationError from '../utils/sendApiValidationError.js';
 import { getAverageRating, getUserRating } from '../services/ratingService.js';
-import { GammaUser } from '../models/gammaModels.js';
+import sendApiValidationError from '../utils/sendApiValidationError.js';
 
 const gameRouter = Router();
 
+const isInt = /^\d+$/;
+const intMessage = { message: 'Not an integer.' };
+const gamesQuerySchema = z
+	.object({
+		search: z.string().min(1).max(500).optional(),
+		platform: z.string().min(1).optional(),
+		releaseBefore: z.string().datetime().optional(), // ISO date string
+		releaseAfter: z.string().datetime().optional(), // ISO date string
+		playtimeMin: z.string().min(1).regex(isInt, intMessage).optional(),
+		playtimeMax: z.string().min(1).regex(isInt, intMessage).optional(),
+		playerCount: z
+			.string()
+			.min(1)
+			.max(2000)
```

Comment:
Why does only `playerCount` have a max number of chars? And if we have a max I then a number with 2000 chars is quite big...

Reply:
That was probably a result of me just chaning the type to string 😅 Will fix. Yeah 2000 is a bit big. Probably more reasonable to have like 4 or something